source/error_report/src/main.rs

```diff
@@ -0,0 +1,221 @@
+use chrono::{prelude::*, DateTime};
+use std::time::{Duration, Instant};
+use std::{
+    env,
+    fs::{self, File},
+    io::prelude::*,
+    io::{BufRead, BufReader},
+    path::Path,
+    process::{Command, Stdio},
+    str,
+};
+use toml::{map::Map, value::Value};
+use zip::write::FileOptions;
+
+fn main() {
+    let mut file_path = String::new();
+    #[allow(unused_assignments)]
+    let mut our_args = Vec::new();
+
+    let args: Vec<String> = env::args().collect();
+    if args.len() > 1 {
+        for argument in &args {
+            if argument.ends_with(".rs") {
+                file_path = argument.clone();
+            }
+        }
+        our_args = args[2..].to_vec();
+    } else {
+        println!("Usage: verus --error-report <args..>");
+        return;
+    }
+
+    // assumption: when verus is invoking error_report, the dir of verus path should be put in the first argument
+    let program_dir = args[1].clone();
+
+    let z3_path = Path::new(&program_dir).join("z3");
+    let verus_path = Path::new(&program_dir).join("verus");
+
+    let mut verus_call = our_args.clone();
+    verus_call.insert(0, verus_path.to_string_lossy().to_string());
+
+    let z3_version_output =
+        Command::new(z3_path).arg("--version").output().expect("failed to execute process");
+    let verus_version_output =
+        Command::new(&verus_path).arg("--version").output().expect("failed to execute process");
+
+    // mandating --time flag, we remove --time flag here to prevent two --time flags that panic verus
+    if our_args.contains(&"--time".to_string()) {
+        our_args.retain(|x| x != "--time");
+    }
+
+    let start_time = Instant::now();
+    let child = Command::new(verus_path)
+        .stdin(Stdio::null())
+        .args(our_args)
+        .arg("--emit=dep-info")
+        .arg("--time")
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .spawn()
+        .expect("failed to execute process");
+    let verus_output: std::process::Output =
+        child.wait_with_output().expect("Failed to read stdout");
+    let verus_duration = start_time.elapsed();
+
+    // The following method calls do the actual work of writing a toml file
+    // with relevant information and saving the toml file and relevant files
+    // to a zip file
+    toml_setup_and_write(
+        verus_call,
+        z3_version_output,
+        verus_version_output,
+        verus_output,
+        verus_duration,
+    );
+    let (dependencies_path, zip_path) = zip_setup(file_path);
+    println!("Stored error report to {}\n", zip_path);
+
+    fs::remove_file("error_report.toml").expect("failed to delete toml file\n");
```

Comment:
If we panic between the time these files are created and when this line is reached, they won't be cleaned up (this has happened to me while trying this out). Error handling with `Result` should help with this.